### PR TITLE
Clean up the route step

### LINKED ISSUE #15039: Clean up the route step
### User Story
As a developer I want to prune unused/old code and related documentation so the codebase stays clean.

### Description/Use Case
We recently split the route step into two new steps. We need to remove traces of the old route step from the code.

### Risks/Impacts/Considerations
Make sure not to remove anything that might still be used such as datatypes/objects/etc.

### Dev Notes
Search for `/// TODO: remove after route queue empty` and this ticket number

### Acceptance Criteria
- [ ] `FHIRRouter.kt` and related tests/docs deleted from the repo
- [ ] All TODOs addressed
- [ ] #14904 merged and in production at least one deployment prior to the PR for this ticket getting merged

### PR DESCRIPTION
**Pull request by thetaurean**
# * DO NOT MERGE BEFORE #14904 DEPLOYS *

This PR removes left over code from the route step. 

Test Steps:
1. Run test suites

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [x] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] ~~Added~~ Removed tests?

## Linked Issues
- Fixes #15039 

## To Be Done
- Clean up or remove `FhirFunctionTests` and `FhirFunctionIntegrationTests.kt`

### CODE REVIEW

**Review by JFisk42 — APPROVED** [review]
Love to see a code deletion PR 🤩

File: prime-router/src/main/kotlin/fhirengine/azure/FHIRFunctions.kt [hunk 1]
```diff
@@ -71,43 +69,6 @@ class FHIRFunctions(
...
-        val messagesToDispatch = runFhirEngine(message, dequeueCount, fhirEngine, actionHistory)
-        messagesToDispatch.forEach {
-            queueAccess.sendMessage(
-                elrTranslationQueueName,
-                it.serialize()
-            )
-        }
-    }
-
     /**
```
[the remark this answers is not in the document]
Comment (thetaurean) [reply prime-router/src/main/kotlin/fhirengine/azure/FHIRFunctions.kt:72]: Ooh good catch I'll update that

File: prime-router/src/main/kotlin/azure/ActionHistory.kt [hunk 2]
```diff
@@ -541,12 +543,14 @@ class ActionHistory(
         trackFilteredItems(report)
         trackItemLineages(report)
 
-        // batch queue messages are added by the batchDecider, not ActionHistory
+        // some event actions are tracked in the functions themselves, not ActionHistory.
         // TODO: Need to update this process to have a better way to determine what messages should be sent
         //  automatically as part of queueMessages and what are being send manually as part of the parent function.
         //  The automatic queueing uses the action name as the queue name, and this is not the case for FHIR actions
         if (event.eventAction != Event.EventAction.BATCH &&
             event.eventAction != Event.EventAction.ROUTE &&
```
Comment (arnejduranovic) [inline prime-router/src/main/kotlin/azure/ActionHistory.kt:551]: Should all references to ROUTE be removed as well as removing the EventAction?

File: prime-router/src/test/kotlin/history/SubmissionHistoryTests.kt [hunk 3]
```diff
@@ -692,7 +692,6 @@ class SubmissionHistoryTests {
         }
     }
 
-    // TODO: remove after route queue empty (see https://github.com/CDCgov/prime-reportstream/issues/15039)
```
Comment (mkalish) [inline prime-router/src/test/kotlin/history/SubmissionHistoryTests.kt:695]: Looks like it was maybe intentional to keep this test?  I would maybe leave a comment indicating that it exists to test historical purposes?

File: prime-router/src/test/kotlin/history/azure/SubmissionFunctionIntegrationTests.kt [hunk 4]
```diff
@@ -307,7 +307,6 @@ class SubmissionFunctionIntegrationTests {
         assertThat(historyNode.get("warnings").size()).isEqualTo(1)
     }
 
-    // TODO: remove after route queue empty (see https://github.com/CDCgov/prime-reportstream/issues/15039)
```
Comment (mkalish) [inline prime-router/src/test/kotlin/history/azure/SubmissionFunctionIntegrationTests.kt:310]: Same question here

File: prime-router/src/test/kotlin/history/azure/SubmissionFunctionIntegrationTests.kt
Comment (mkalish) [inline prime-router/src/test/kotlin/history/azure/SubmissionFunctionIntegrationTests.kt:1]: I think this needs to get updated across the board to use the new steps?

File: prime-router/docs/universal-pipeline/route.md
[the remark this answers is not in the document]
Comment (thetaurean) [reply prime-router/docs/universal-pipeline/route.md:1]: Added in the previous PR that added the new steps :).

File: prime-router/src/test/kotlin/history/azure/SubmissionFunctionIntegrationTests.kt
Comment (thetaurean) [reply prime-router/src/test/kotlin/history/azure/SubmissionFunctionIntegrationTests.kt:1]: My workflow was to duplicate the tests that mention the route step.

I think I got all of them but I'll double check.

**Review by mkalish — APPROVED** [review]
👍🏻

**Pull request merged**